plugins/chat/lib/chat/user_option_extension.rb

```diff
@@ -18,8 +18,13 @@ def base.chat_header_indicator_preferences
         @chat_header_indicator_preferences ||= { all_new: 0, dm_and_mentions: 1, never: 2 }
       end
 
-      base.enum :chat_email_frequency, base.chat_email_frequencies, prefix: "send_chat_email"
-      base.enum :chat_header_indicator_preference, base.chat_header_indicator_preferences
+      if !base.method_defined?(:send_chat_email_never?) # Avoid attempting to override when autoloading
+        base.enum :chat_email_frequency, base.chat_email_frequencies, prefix: "send_chat_email"
+      end
+
+      if !base.method_defined?(:never?) # Avoid attempting to override when autoloading
+        base.enum :chat_header_indicator_preference, base.chat_header_indicator_preferences
```

Comment:
@jjaffeux not strictly related to this PR, I think we should probably add a `prefix` to this enum. Right now it's auto-defining three methods:

- `UserOption#never?`
- `UserOption#all_new?`
- `UserOption#dm_and_mentions?`

IMO would be much better to prefix so the methods are something like `#chat_header_indicator_never?`

Reply:
Yes agreed.